source/renderer/app/stores/VotingStore.js

```diff
@@ -433,8 +447,19 @@ export default class VotingStore extends Store {
     }
   };
 
-  @action _checkVotingRegistrationEnd = () => {
-    this.isRegistrationEnded = new Date() >= VOTING_REGISTRATION_END_DATE;
+  @action _checkFundPhase = (now: Date) => {
+    const phaseValidation = {
+      [FundPhases.SNAPSHOT]: (date: Date) => date < VOTING_CAST_START_DATE,
+      [FundPhases.VOTING]: (date: Date) =>
+        now >= VOTING_CAST_START_DATE && date < VOTING_CAST_END_DATE,
+      [FundPhases.TALLYING]: (date: Date) =>
+        now >= VOTING_CAST_END_DATE && date < VOTING_RESULTS_DATE,
+      [FundPhases.RESULTS]: (date: Date) => date >= VOTING_RESULTS_DATE,
+    };
+    this.fundPhase =
+      Object.keys(FundPhases)
```

Comment:
I had to use this:
Object.keys(FundPhases).map((phase) => FundPhases[phase])
Instead of:
Object.values(FundPhases)
Due to flow not inferring the type correctly

facebook/flow#2221

Reply:
I had solved it with a switch case maybe, but your solution is also OK.